include/compartment.h

```diff
@@ -127,11 +129,12 @@ struct LibDependency
     // TLS-related variables
     // TODO can there be more TLS sections?
     void *tls_sec_addr;
+    ptrdiff_t tls_sec_off;
     size_t tls_sec_size;
     size_t tls_data_size;
     // offset from TLS base pointer (i.e., value of `tpidr_el0`) where this
     // library's TLS variables start
-    size_t tls_offset;
+    size_t tls_offset; // TODO consider other type (`void*` or `ptrdiff_t`?)
```

Comment:
I think `size_t` here is correct: `ptrdiff_t` is signed which I don't think you want.

Reply:
Yup, looking again over it, you're right. Removed in 2830554bfc524a150fed656cc8269fb124ab0602.